backend/functions/student/functions.js

```diff
@@ -36,18 +45,26 @@ const addStudentSurveyResponse = async (
   const crseIds = await Promise.all(
     courseCatalogNames.map((name) => mapCatalogNameToCrseId(name))
   );
+  const ref = studentRef
+  .doc(email)
+  const existing_crses = (await ref.get()).data().groups;
```

Comment:
`.data()` is undefined when the document doesn't exist (i.e. this email has never submitted a survey), so getting groups from it throws an error